lua/nvim-tree/git/init.lua

```diff
@@ -22,25 +22,36 @@ function M.reload()
   return M.projects
 end
 
-function M.reload_project(project_root)
+function M.reload_project(project_root, path)
   local project = M.projects[project_root]
   if not project or not M.config.enable then
     return
   end
 
-  local watcher = M.projects[project_root].watcher
-  M.projects[project_root] = {}
+  if path and not path:match("^" .. project_root) then
```

Comment:
how is this even possible ?

Reply:
How is "no match" possible? Not sure...

I'd prefer to have this safety measure in place, as falling back to a full refresh is better than having the git invocation fail.

Reply:
what i meant is that if the path does not match, it will not load a project and thus return early.

Reply:
Ah I see... if there is no match, the update for that directory would be meaningless.

I'll change in the next PR, to keep things simple.